apps/vmq_server/src/vmq_subscriber_db.erl

```diff
@@ -59,17 +59,23 @@ subscribe_db_events() ->
           when (Val == ?TOMBSTONE) or (Val == undefined) ->
             ignore;
         ({deleted, ?SUBSCRIBER_DB, SubscriberId, Subscriptions}) ->
-            Deleted = vmq_subscriber:get_changes(Subscriptions),
+            Deleted = vmq_subscriber:get_changes(get_right_subs(Subscriptions)),
             {delete, SubscriberId, Deleted};
         ({updated, ?SUBSCRIBER_DB, SubscriberId, OldVal, NewSubs})
           when (OldVal == ?TOMBSTONE) or (OldVal == undefined) ->
-            Added = vmq_subscriber:get_changes(NewSubs),
+            Added = vmq_subscriber:get_changes(get_right_subs(NewSubs)),
             {update, SubscriberId, [], Added};
         ({updated, ?SUBSCRIBER_DB, SubscriberId, OldSubs, NewSubs}) ->
-            {Removed, Added} = vmq_subscriber:get_changes(OldSubs, NewSubs),
+            {Removed, Added} = vmq_subscriber:get_changes(get_right_subs(OldSubs), get_right_subs(NewSubs)),
             {update, SubscriberId, Removed, Added};
         (_) ->
             ignore
     end.
 
+get_right_subs(Subs) when is_list(Subs) ->
+    Subs;
+get_right_subs(Subs) ->
```

Comment:
This is what I get on my machine with 0.15.2 which is a list and not a tuple.
```erlang
{application_start_failure,vmq_server,
  {{undef,[{vmq_subscriber,change_node_all,
    [[{'VerneMQ1@127.0.0.1',false,[{[<<\"a\">>,<<\"b\">>,<<\"+\">>],1}]}],
    'VerneMQ2@127.0.0.1'],[]},{vmq_server_sup,'-maybe_change_nodename/0-fun-2-',3,
...
```
Is it possible that the same node has 'old' subscriptions in the old format?